gst/nnstreamer/tensor_common.c

```diff
@@ -1235,6 +1235,48 @@ gst_tensors_caps_from_config (const GstTensorsConfig * config)
   return gst_caps_simplify (caps);
 }
 
+/**
+ * @brief Get possible caps for given config
+ * @param config tensor config info
+ * @return GstCaps* all possible caps
+ */
+GstCaps *
+gst_tensors_possible_caps_from_config (const GstTensorsConfig * config)
+{
+  GstCaps *caps;
+
+  g_return_val_if_fail (config != NULL, NULL);
+
+  caps = gst_caps_new_empty ();
+
+  /* allow other/tensor in sink caps for backward compability */
+  if (config->info.num_tensors == 1) {
+    GstTensorConfig c;
+
+    gst_tensor_config_init (&c);
+    c.info = config->info.info[0];
+    c.rate_n = config->rate_n;
+    c.rate_d = config->rate_d;
+
+    gst_caps_append (caps, gst_tensor_caps_from_config (&c));
+  }
+
+  /* other/tensors */
+  gst_caps_append (caps, gst_tensors_caps_from_config (config));
+
+  /* other/tensors-flexible */
+  if (!gst_tensors_info_is_flexible (&config->info)) {
```

Comment:
Is it correct?
```suggestion
  if (gst_tensors_info_is_flexible (&config->info)) {
```

Reply:
Yes. Same code copied from `tensor_filter.c` which always appends flexible caps

Reply:
Ah.. Thanks for the explanation.